entity_recognition/entity_recognition_training.ipynb

```diff
@@ -0,0 +1,717 @@
+    "- Compute out-of-sample predicted probabilities via cross-validation \n",
+    "- Reduce subword-level tokens to word-level tokens \n",
+    "\n",
+    "\\* In most NLP literatures, tokens typically refer to words or punctuation marks, while most modern tokenizers break down longer words into subwords. To avoid confusion, given tokens are referred as \"word-level tokens\", which represent the individual tokens given in the original dataset with a separate class label provided for each token; tokens obtained from the tokenizers are referred as \"subword-level tokens\". "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1da020bc",
+   "metadata": {},
+   "source": [
+    "## 1. Install the required dependencies "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d9e0fb4a",
+   "metadata": {},
+   "source": [
+    "You can use `pip` to install all packages required for this tutorial as follows: \n",
+    "\n",
+    "    !pip install tqdm transformers \n",
+    "    !pip install cleanlab "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9018de0f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!wget -nc https://data.deepai.org/conll2003.zip && mkdir data \n",
+    "!unzip conll2003.zip -d data/ && rm conll2003.zip "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a27944bd",
+   "metadata": {
+    "nbsphinx": "hidden"
+   },
+   "outputs": [],
+   "source": [
+    "# Package installation (hidden on docs website).\n",
+    "# Package versions used: tqdm==4.64.0 transformers==4.22.0.dev0 cleanlab==2.0.0 numpy==1.23.0 sklearn==0.0 \n",
+    "# (cont): torch==1.12.0 nltk==3.7 pytorch_transformers==1.2.0 seqeval==1.2.2 \n",
+    "# ericwang/cleanlab -b token_classification for now \n",
+    "\n",
+    "dependencies = [\"tqdm\", \"transformers\", \"cleanlab\", \"sklearn\", \"torch\", \"nltk\", \"pytorch_transformers\", \"seqeval\"]\n",
+    "\n",
+    "if \"google.colab\" in str(get_ipython()):  # Check if it's running in Google Colab\n",
+    "    %pip install cleanlab  # for colab\n",
+    "    cmd = ' '.join([dep for dep in dependencies if dep != \"cleanlab\"])\n",
+    "    %pip install $cmd\n",
+    "else:\n",
+    "    missing_dependencies = []\n",
+    "    for dependency in dependencies:\n",
+    "        try:\n",
+    "            __import__(dependency)\n",
+    "        except ImportError:\n",
+    "            missing_dependencies.append(dependency)\n",
+    "\n",
+    "    if len(missing_dependencies) > 0:\n",
+    "        print(\"Missing required dependencies:\")\n",
+    "        print(*missing_dependencies, sep=\", \")\n",
+    "        print(\"\\nPlease install them before running the rest of this notebook.\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a1349304",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# !pip install transformers tqdm \n",
+    "import numpy as np\n",
```

Comment:
[nit] Ideally these imports will be in a separate cell from the installation stuff above